realm/src/main/java/io/realm/RealmObject.java

```diff
@@ -103,8 +103,13 @@ public void removeFromRealm() {
      * @return {@code true} if the object is still accessible, {@code false} otherwise or if it is a
      * standalone object.
      */
-    public boolean isValid() {
-        return row != null && row.isAttached();
+    public boolean
+    isValid() {
+        try {
+            return row != null && row.isAttached();
+        } catch (IllegalStateException e) {
+            return false;
```

Comment:
It would be much better to change InvalidRow so it returned `false`. Using exceptions to control flow is very bad for performance.